File: javascript/src/exchange-code-for-tokens.js
import config from './config.js';
import { Moneyhub } from '@mft/moneyhub-api-client';
import yargs from 'yargs';
import { hideBin } from 'yargs/helpers';

const args = yargs(hideBin(process.argv))
  .option('code', {
    alias: 'c',
    describe: 'code to exchange code for tokens',
  })
  .option('nonce', {
    alias: 'n',
    describe: 'nonce from original authorisation URL',
    default: 'bar',
  })
  .option('id_token', {
    alias: 'i',
    describe: 'id_token from redirect url',
  })
  .option('state', {
    alias: 's',
    describe: 'state from original authorisation URL',
    default: 'foo',
  }).argv;

const main = async (code) => {
  const moneyhub = await Moneyhub(config);

  const tokens = await moneyhub.exchangeCodeForTokens({
    localParams: {
      state: args.state,
      nonce: args.nonce,
    },
    paramsFromCallback: {
      code: args.code,
      state: args.state,
      id_token: args.id_token
    },
  });

  console.log(tokens);
};

main();

File: python/src/exchange_code_for_tokens.py
from config import *
import utils
import argparse

parser = argparse.ArgumentParser(description='Get AIS authorisation URL')
parser.add_argument("-s", "--state", default="foo", help="The state value")
parser.add_argument("-n", "--nonce", default="bar", help="The nonce value")
parser.add_argument("-c", "--code", help="The code to exchange tokens for")
parser.add_argument("-i", "--id_token", help="The code to exchange tokens for")
args = parser.parse_args()
identity_server = config["identity_service_url"]

utils.verify_id_token(config=config, id_token=args.id_token, nonce=args.nonce)
tokens = utils.exchange_code_for_tokens(config, args.nonce, args.id_token, args.code)

print(tokens)